examples/Generic/README.md

```diff
@@ -2,21 +2,160 @@
 
 This directory contains a minimal example that shows how to use `AstarteGenericDevice`.
 
-The device is instantiated in `ExampleDevice.java`, the code is commented to highlight all the
-things needeed to create your Astarte Device.
+The device is instantiated in `ExampleDevice`, the code is commented to highlight all the
+things needed to create your Astarte Device.
+
+
+## Downloading
+
+Add to "dependencies" (as for any module you need):
+
+``` groovy
+dependencies {
+    implementation 'org.astarte-platform:devicesdk-generic:$version'
+}
+```
+
+Usage
+-----
+
+### 1. Device registration and credentials secret emission
+
+*If your device is already register, you can skip this section.*
+
+You must register the device beforehand to obtain its `credentials-secret`.
```

Comment:
I think that we are preferring "The device must be registered" rather than writing "You must ...".